h/api/resources.py

```diff
@@ -0,0 +1,45 @@
+# -*- coding: utf-8 -*-
+
+from pyramid.security import Allow, Authenticated, ALL_PERMISSIONS
+
+from .models import Annotation
+
+
+class Resource(dict):
+    """
+    Resource is an entry in the traversal tree.
+    """
+    __name__ = None
+    __parent__ = None
+
+    def __init__(self, request):
+        pass
+
+    def add(self, name, obj):
+        obj.__name__ = name
+        obj.__parent__ = self
+        self[name] = obj
+
+
+class Annotations(object):
+    """
+    Annotations is a container resource that exposes annotations as its
+    children in the tree.
+    """
+    def __getitem__(self, key):
+        instance = Annotation.fetch(key)
+        if instance is None:
+            raise KeyError(key)
+        instance.__name__ = key
+        instance.__parent__ = self
+        return instance
+
+
+class Root(Resource):
+    __acl__ = [
+        (Allow, 'group:admin', ALL_PERMISSIONS),
+        (Allow, Authenticated, 'create'),
+    ]
+
+    def __init__(self, request):
+        self.add('annotations', Annotations())
```

Comment:
It might be because I'm not overly familiar with how most people structure traversal trees, but I don't really understand the advantages of disadvantages of doing it either the way I've done it or the way you propose. 

Here's how I see it: `Annotations` isn't a `Resource` because it doesn't have an `add` method -- its traversal children are already fully defined -- they're all the annotations, and that's why it has an explicit `__getitem__` method.

For the explicit factory function I think that makes a bit more sense -- you're getting at the fact that the request parameter isn't used, right? I'll do that, I think.